File: EULER/39_integer_right_triangles.cpp
#include <bits/stdc++.h>
using namespace std;


int main(){
  vector<int> freqs(1001, 0);
  for (int p=12; p<1001; p++){
    for (int a=3; a<334; a++){
      for (int c=a+1; c<p-2*a+1; c++){
        int b_exp_sq = (p - a - c) * (p - a - c);
        int real_b_sq = c * c - a * a;
        if (b_exp_sq == real_b_sq) freqs[p]++;
      }
    }
  }
  int pos = max_element(freqs.begin(), freqs.end()) - freqs.begin();
  cout << pos << endl;
}

File: EULER/39_integer_right_triangles.py
freqs = [0 for _ in range(1001)]
ans_freq, ans_val = 0, 0

for p in range(12, 1001):
  for a in range(3, 334):
    for c in range(a+1, p-2*a+1):
      b_exp_sq = (p - a - c) ** 2
      real_b_sq = c**2 - a**2
      if b_exp_sq == real_b_sq:
        freqs[p] += 1
        if freqs[p] > ans_freq:
          ans_freq, ans_val = freqs[p], p

print(ans_val)
